src/canvas-element-bitmap-size.ts

```diff
@@ -23,17 +28,29 @@ export interface Binding extends Disposable {
 	subscribeSuggestedBitmapSizeChanged(listener: SuggestedBitmapSizeChangedListener): void;
 	unsubscribeSuggestedBitmapSizeChanged(listener: SuggestedBitmapSizeChangedListener): void;
 	applySuggestedBitmapSize(): void;
+
+	get2DContext(contextOptions?: CanvasRenderingContext2DSettings): Canvas2DContext<OffscreenAllowed> | null;
+
+	readonly usingOffscreenCanvas: boolean;
+	requestOffscreenCanvas(): OffscreenCanvas | null;
+	releaseOffscreenCanvas(canvas: OffscreenCanvas): void;
 }
 
 export interface DevicePixelContentBoxBindingTargetOptions {
 	allowResizeObserver?: boolean;
+	/**
+	 * If `true` fancy canvas will attempt to use OffscreenCanvas
+	 * (if supported by the browser).
+	 */
+	allowOffscreenCanvas?: boolean;
 }
 
-class DevicePixelContentBoxBinding implements Binding, Disposable {
+class DevicePixelContentBoxBinding<OffscreenAllowed extends boolean> implements Binding<OffscreenAllowed>, Disposable {
```

Comment:
Let's try to decouple `OffscreenCanvas` size control logic from the `DevicePixelContentBoxBinding`.
In fact, only `HTMLCanvasElement` cares about its content box and occupied space on the screen to sync the bitmap size with content box. `OffscreenCanvas` bitmap could have any size, for example two times larger or smaller than the target canvas' bitmap.

I think it's more flexible to sync `OffscreenCanvas` bitmap size with `HTMLCanvasElement` bitmap size in a separate binding. It should be something like a CHAIN of bindings: `HTMLCanvasElement` bitmap size is bound with his own content box by `DevicePixelContentBoxBinding`, and `OffscreenCanvas` bitmap size is bound with target `HTMLCanvasElement` bitmap size by another binding, what we are going to implement in this PR)

So the size change propagation chain should look like: CanvasElement's `device-pixel-content-box` ==> CanvasElement's bitmap size ==> `OffscreenCanvas` bitmap size. And here enters the flexibility: we could subscribe MULTIPLE `OffscreenCanvas`es to the same CanvasElement, we could introduce some size transform in-between Element and Offscreen (to perform super- or subsampling) and so on.

I'll come back later with some suggestions on naming and module structure.